apis/python/src/tiledbsoma/io/ingest.py

```diff
@@ -1394,6 +1425,52 @@ def _ingest_uns_node(
     logging.log_io(msg, msg)
 
 
+def _ingest_uns_string_array(
+    coll: AnyTileDBCollection,
+    key: str,
+    value: NPNDArray,
+    platform_config: Optional[PlatformConfig],
+    context: Optional[SOMATileDBContext],
+    *,
+    use_relative_uri: Optional[bool],
+    ingestion_params: IngestionParams,
+) -> None:
+    """
+    Ingest an uns string array. In the SOMA data model, we have NDArrays _of number only_ ...
+    so we need to make this a SOMADataFrame.
+
+    Ideally we don't want to an index column "soma_joinid" -- "index", maybe.
+    However, ``SOMADataFrame`` _requires_ that soma_joinid be present, either
+    as an index column, or as a data column. The former is less confusing.
+    """
+    if len(value.shape) != 1:
+        msg = (
+            f"Skipped {coll.uri}[{key!r}]"
+            f" (uns object): string-array is not one-dimensional"
+        )
+        logging.log_io(msg, msg)
+        return
+
+    n = len(value)
+    df_uri = _util.uri_joinpath(coll.uri, key)
+    df = pd.DataFrame(
+        data={
+            "soma_joinid": np.asarray(range(n), dtype=np.int64),
+            "values": [str(e) for e in value],
```

Comment:
If this is nullable, this would turn `None` into `"None"`. Is this what we want? (If the dataframe is non-nullable, `str(e) if e else ""` might be a better formulation, so that Nones become the empty string.)